### DISCUSSION TITLE
Trim doesn't call with pagy_nav_js when first page is active

### DISCUSSION THREAD

**Discussion by millefiori** [ddnexus/pagy#655] [Q&A] [answered]
On pagy_bootstrap_nav_js. When page=1 is active,  item is a String `item === "1"`. Then this code doesn't call trim. This is performance issue?

https://github.com/ddnexus/pagy/blob/master/lib/javascripts/pagy-dev.js#L32

**Comment by ddnexus** [comment, marked as answer]
You mean that the bootstrap trimming is skipped on page 1 when it's active.

That works for all the framework extras because either they have a `a` tag without `href` or `href="#"` or they are disabled links, or they don't even have a link for the active page. Bootstrap has an active and complete URL to the active page which seems wrong!

The way trimming works is that usually you have the first page url called without any page param (it's implicit page 1 by default), when you are in any other page, the link to page 1 gets trimmed (since the page is not the active one, and clicking on it you want to remove page 1). Even in bootstrap, if you go to page 2, then click on the link of page 1, you don't see the `page=1` param in the address bar. But if you click on page 1 again being already at page 1 you reload the same page, this time with the added `page=1`.

That is a bug, because it should not happen. But the bug seems to be on the HTML side: the page should not be linked to itself at all. Then we could even make the javascript more robust for any framework that does not behave like we would expect. We should also remove the `page=1` in the rare case the `page=1` is in the url, regardless active or not.

Thank you for reporting it! I created an issue out of this!

**Reply by ddnexus** [reply]
Fixed in 7.0.6
